runtime/compiler/trj9/il/J9SymbolReference.cpp

```diff
@@ -258,35 +258,38 @@ SymbolReference::getTypeSignature(int32_t & len, TR_AllocationKind allocKind, bo
                   !comp->getOption(TR_AOT) &&
                    (type == TR::Address)))
                  {
-                 void * p = symbol->getStaticAddress();
                  TR::VMAccessCriticalSection vmAccessCriticalSection(comp->fej9(),
                                                                       TR::VMAccessCriticalSection::tryToAcquireVMAccess,
                                                                       comp);
-                 if ((*(void **)p != 0) &&
-                     vmAccessCriticalSection.hasVMAccess())
+                 if (vmAccessCriticalSection.hasVMAccess())
                     {
-                    TR_OpaqueClassBlock *classOfObject = comp->fej9()->getClassFromStatic(p);
-                    const char * s = TR::Compiler->cls.classNameChars(comp, classOfObject, len);
-                    if (s && (s[0] != '['))
+                    uintptrj_t objectStaticAddress = (uintptrj_t)symbol->getStaticAddress();
+                    uintptrj_t objectRef = comp->fej9()->getStaticReferenceFieldAtAddress(objectStaticAddress);
+                    if (objectRef != 0)
                        {
-                       s = classNameToSignature(s, len, comp);
-                       }
-                    else
-                       {
-                       int32_t numParens = 0;
-                       while (s && (s[0] == '[') && (s[1] == 'L'))
+                       TR_OpaqueClassBlock *classOfObject = comp->fej9()->getClassFromStatic((void *)objectRef);
```

Comment:
Should this be `getClassFromStatic((void*)objectStaticAddress)`?

Reply:
No; previously getClassFromStatic would dereference the object location. However, because the call to getClassFromStatic requires checking if the dereference is NULl, I changed getClassFromStatic to take in the dereferenced object pointer instead (https://github.com/eclipse/openj9/pull/1260/files#diff-cba9fe3d00eac2598c34d768e9173980R8516)

Reply:
In that case `getClassFromStatic` might now be redundant with `getObjectClass`

Reply:
Hm I think you're right. I guess I can get rid of `getClassFromStatic` and replace the contents of `getObjectClass` with `J9OBJECT_CLAZZ`